src/gameLogic/execution/gadget/LaserCompact.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+* @file   LaserCompact.cpp
+* @author Marco Deuscher
+* @date   04.05.2020 (creation)
+* @brief  Implementation of laser compact gadget execution
+*/
+
+#include "GadgetExecutor.hpp"
+#include "util/GameLogicUtils.hpp"
+
+namespace spy::gameplay {
+
+    bool GadgetExecutor::executeLaserCompact(State &s, const GadgetAction &a, const MatchConfig &config) {
+        auto character = s.getCharacters().getByUUID(a.getCharacterId());
+        // check if shot hit
+        bool hitSuccessfull = util::GameLogicUtils::probabilityTestWithCharacter(s, *character,
+                                                                                 config.getLaserCompactHitChance());
+
+        if (!hitSuccessfull) {
+            return false;
+        }
+
+        // check if person on field
+        bool personOnField = util::GameLogicUtils::isPersonOnField(s, a.getTarget());
+
+        if (personOnField) {
+            auto targetPerson = util::GameLogicUtils::getInCharacterSetByCoordinates(s.getCharacters(), a.getTarget());
+            targetPerson->removeGadget(gadget::GadgetEnum::COCKTAIL);
+        } else {
+            // remove cocktail from field
+            s.getMap().getField(a.getTarget()).removeGadget();
```

Comment:
Der Klingen-Hut z.B. kann auch auf einem Feld liegen, der sollte nicht removed werden. Wird das im validate schon geprüft? Wenn ja wäre hier kurzer kommentar gut

Reply:
In der validate wird bereits sicher gestellt, dass es auf dem Feld einen Cocktail gibt. Entweder auf dem Tisch oder in der Hand.